crates/router/src/connector/threedsecureio.rs

```diff
@@ -0,0 +1,581 @@
+    for Threedsecureio
+{
+}
+
+impl ConnectorIntegration<api::Capture, types::PaymentsCaptureData, types::PaymentsResponseData>
+    for Threedsecureio
+{
+}
+
+impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsResponseData>
+    for Threedsecureio
+{
+}
+
+impl ConnectorIntegration<api::Execute, types::RefundsData, types::RefundsResponseData>
+    for Threedsecureio
+{
+}
+
+impl ConnectorIntegration<api::RSync, types::RefundsData, types::RefundsResponseData>
+    for Threedsecureio
+{
+}
+
+#[async_trait::async_trait]
+impl api::IncomingWebhook for Threedsecureio {
+    fn get_webhook_object_reference_id(
+        &self,
+        _request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<api::webhooks::ObjectReferenceId, errors::ConnectorError> {
+        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+    }
+
+    fn get_webhook_event_type(
+        &self,
+        _request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<api::IncomingWebhookEvent, errors::ConnectorError> {
+        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+    }
+
+    fn get_webhook_resource_object(
+        &self,
+        _request: &api::IncomingWebhookRequestDetails<'_>,
+    ) -> CustomResult<Box<dyn masking::ErasedMaskSerialize>, errors::ConnectorError> {
+        Err(errors::ConnectorError::WebhooksNotImplemented).into_report()
+    }
+}
+
+impl api::ConnectorPreAuthentication for Threedsecureio {}
+impl api::ExternalAuthentication for Threedsecureio {}
+impl api::ConnectorAuthentication for Threedsecureio {}
+impl api::ConnectorPostAuthentication for Threedsecureio {}
+
+impl
+    ConnectorIntegration<
+        api::Authentication,
+        types::authentication::ConnectorAuthenticationRequestData,
+        types::authentication::AuthenticationResponseData,
+    > for Threedsecureio
+{
+    fn get_headers(
+        &self,
+        req: &types::authentication::ConnectorAuthenticationRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, request::Maskable<String>)>, errors::ConnectorError> {
+        self.build_headers(req, connectors)
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        self.common_get_content_type()
+    }
+
+    fn get_url(
+        &self,
+        _req: &types::authentication::ConnectorAuthenticationRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Ok(format!("{}/auth", self.base_url(connectors),))
+    }
+
+    fn get_request_body(
+        &self,
+        req: &types::authentication::ConnectorAuthenticationRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<RequestContent, errors::ConnectorError> {
+        let connector_router_data = threedsecureio::ThreedsecureioRouterData::try_from((
+            &self.get_currency_unit(),
+            req.request
```

Comment:
This would be the common flow for both payment and non-payment authentication (payment_method authentication). payment_method authentication is yet to be implemented, which doesn't have amount and currency in the authentication flow